### DISCUSSION TITLE
pywinkeyer problems and questions

### DISCUSSION THREAD

**Discussion by awandahl** [mbridak/not1mm#137] [General]
PARTLY SOLVED BY WORK-AROUND, SEE MY COMMENT!

I can run https://github.com/mbridak/PyWinKeyerSerial properly (on an Arduino Winkeyer clone) but I have problems with the pywinkeyer. Is it this one I should use? https://github.com/drewarnett/pywinkeyerdaemon  I have tried that one, changed it for proper unicode handling, but get this:  

```
aw@5x7w-t2:~$ ./pywinkeyerdaemon.py -d /dev/ttyUSB0
Traceback (most recent call last):
  File "/home/aw/./pywinkeyerdaemon.py", line 777, in <module>
    winkeyer = WinKeyer(args.device, debug=args.debug)
  File "/home/aw/./pywinkeyerdaemon.py", line 99, in __init__
    self.host_open()
  File "/home/aw/./pywinkeyerdaemon.py", line 126, in host_open
    raise IOError("No response from WinKeyer device")
OSError: No response from WinKeyer device
```
I run on a Tinkerboard 2 (Rockchip):
```
aw@5x7w-t2:~$ cat /etc/os-release
PRETTY_NAME="Armbian 24.8.2 jammy"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.armbian.com"
SUPPORT_URL="https://forum.armbian.com"
BUG_REPORT_URL="https://www.armbian.com/bugs"
PRIVACY_POLICY_URL="https://www.armbian.com"
UBUNTU_CODENAME=jammy
ARMBIAN_PRETTY_NAME="Armbian 24.8.2 jammy"
aw@5x7w-t2:~$
```
```
aw@5x7w-t2:~$ lsusb
....
Bus 001 Device 008: ID 1a86:7523 QinHeng Electronics CH340 serial converter
...
```
```
aw@5x7w-t2:~$ ls -la /dev/ttyUSB*
crw-rw---- 1 root dialout 188, 0 Sep 17 10:31 /dev/ttyUSB0

```
```
aw@5x7w-t2:~$ groups aw
aw : aw tty disk dialout sudo audio video plugdev games users systemd-journal input netdev bluetooth
```
Any hints would be welcome...

Anders SM0HPL  5X7W

**Comment by awandahl** [comment]
Commenting myself: I installed https://github.com/ok2cqr/winkeyer_server instead and it's running on port 6789. It found my K3NG arduino winkeyer without any problems. By choosing "cwdaemon" in the not1mm configuration I can now transmit fine, and I can also change the speed from the not1mm interface. What I can't do is (of course) to use the speed control on the actual K3NG device. Another thing that struck me is that not1mm lacks the ESM function (https://n1mmwp.hamdocs.com/setup/function-keys/#esm-enter-sends-message) which in my view is the first thing you activate in N1MM. I think I've seen requests for this feature somewhere else in the forum. A big step forward for me. No joy with pywinkeyerdaemon.py, but if ESM is activated I would be up and running and ready to take on some contests...

**Comment by mbridak** [comment]
@awandahl Addressing serial devices: I always use the /dev/serial/by-id/whatever-your-device-shows-as as the device to talk to since it can't get re-enumerated to a random number on the next boot if you have more than one serial device.

ESM: I'll think about it. But seems like a lot of trouble to implement for what you get in return.
